src/components/Executions/Tables/contexts.ts

```diff
@@ -1,6 +1,14 @@
 import * as React from 'react';
-import { NodeExecutionsTableState } from './types';
+import {
+    NodeExecutionColumnDefinition,
+    NodeExecutionsTableState
+} from './types';
+
+export interface NodeExecutionsTableContextData {
+    columns: NodeExecutionColumnDefinition[];
+    state: NodeExecutionsTableState;
+}
 
 export const NodeExecutionsTableContext = React.createContext<
-    NodeExecutionsTableState
->({} as NodeExecutionsTableState);
+    NodeExecutionsTableContextData
+>({} as NodeExecutionsTableContextData);
```

Comment:
While I see that this follows the code that existed here before this change, should we consider giving this a default implementation instead?

Reply:
I debated this with myself. Providing a default value gives a fallback for scenarios where a value isn't explicitly provided. These contexts are intended to _only_ be provided in the view where they are being used, effectively scoping the "caching" to only that instance of the view and creating a new context every time.
I'm concerned that failing to provide a scoped context will result in unintended side effects (high memory usage, stale entries) because the objects live around across multiple instantiations of the view.
So, the intention here is that if it is used incorrectly, it will be quite obvious by the errors thrown while attempting to access properties of an empty object.

Alternatively, I could provide a default value that throws/warns upon access so that it's clear that you've hit the default context when that was not expected?

Reply:
Thinking about it some more, and reading some reasoning from others on the same idea, I'm going to leave it as-is and allow the resulting errors to serve as a warning that there needs to be a useful value provided somewhere in the tree for these contexts.